### DISCUSSION TITLE
How to publish once and all subscribers can receive?

### DISCUSSION THREAD

**Discussion by WarGreymon1998** [OpenDDS/OpenDDS#3975] [General]
Greetings,

I am a beginner of OpenDDS. Recently, I tested on three hosts (two virtual machines with Ubuntu and one raspberry pi board with Raspberry Pi OS). I start the Subscriber on two of them and the Publisher on the third. But why only one can receive messages? What can I do to make the Publisher run only once and all Subscribers of the same domain can receive?

I have looked through all the issues and found no suitable answers. I am currently reading "OpenDDS Developer's Guide". But my mother tongue is not English, so the speed is very slow.

Maybe my expression is strange. Thank you for taking the time to read my question, looking forward to your reply. Or let me know if there are similar issues in the list.

Sincerely,
Wang.

**Comment by simpsont-oci** [comment]
It's difficult to determine your problem without more information, but if you know in advance that you will have two subscribers and they both need to receive reliable data, you can wait for the publication matched status on the publisher to show a current match count of two before writing samples, as well having the publisher call call `wait_for_acknowledgements()` before exiting. If you're not sure of the number of subscribers, you can also look into using the durable qos policy to ensure that late-joining subscribers get previously written samples.

If could also possibly be a problem with your network or your firewall preventing data samples from reaching the required destinations.
